Functionality so that a shift cannot be deleted

vubo opened the pull request:
Issue https://github.com/systers/vms/issues/65

Add functionality so that a shift cannot be deleted if at least one volunteer is signed up for it or has logged hours for it.

Functions have been supplemented:
- delete_shift in services.py
- test_delete_shift in tests.py

vubo commented:
I do not understand why, but github shows changes to the files incorrect https://github.com/systers/vms/pull/66/files. It accepts my changes as a new code :open_mouth: 

@tapasweni-pathak do you know why this would happen? Thanks!

tapasweni-pathak commented:
@vubo Can you rebase and go through the log? Don't edit it while rebasing just see the diffs.

You can always reflog and then rebase things. If you didn't find any issues let me know.

tapasweni-pathak commented:
Any luck so far @vubo in finding the issue?

vubo commented:
@tapasweni-pathak I have not yet found the reason for this.

Here is the log for the shiftDelete branch:

```
commit d92685ae67eefecb77140d328d7b048bca72544f
Author: vubo <vubogdanova@gmail.com>
Date:   Thu Jun 25 01:34:02 2015 +0200

    Functionality so that a shift cannot be deleted

commit e6c1ee21e72c1ea47f51f7178e5afd70ff17a183
Merge: f568aea 04d7b0b
Author: Tapasweni Pathak <tapaswenipathak@gmail.com>
Date:   Tue Jun 16 20:44:48 2015 +0530

    Merge pull request #58 from vubo/testJobFin

    Fixed import in the job/tests.py

commit 04d7b0b972adf90c7995543b5ad535cc9d05425d
Author: vubo <vubogdanova@gmail.com>
Date:   Tue Jun 16 16:31:01 2015 +0200

    Fixed import
```

tapasweni-pathak commented:
@vubo No not the log messages, but the diffs. Can you use git rebase --interactive hashcode^. This will open all your commits starting from the hashcode and show the message and diffs. Don't change the options to edit or anything, just go through the diffs your commit have and check if that is fine or not.

vubo commented:
@tapasweni-pathak I did 
```
git rebase --interactive e6c1ee21e72c1ea47f51f7178e5afd70ff17a183
```
and see now:
![commit](https://cloud.githubusercontent.com/assets/7673323/8543960/7f552842-24a1-11e5-8a30-20875393f6cd.png)

How can I find the diffs?

Thank you!

tapasweni-pathak commented:
I'm sorry, I said not to change that p to e, you need to. Then when you load, you will be able to see your diffs, below the commit message. If it is fine there, then we will search for does GitHub have some issues or what. 

@exploreshaifali you faced this issues sometime back, right? GitHub was taking your previous changes as new commits, how did you resolved it?

exploreshaifali commented:
I faced a bit similar problem, my previous merged changes were also coming as new code. 

Reason - The PR was not requested within corrected pair of branches. 

This is what happened with me:
develop branch(into which my code need to be merged)
B1 (branch from which previous code mrged)
B2 (current branch from which I was requesting PR)

So now the PR should be from B2 to develop, but it was from B2 to B1 instead. That was causing problem.

In your case I could see `develop` branch so I think you need to check request branch once, if it is correct or not.

vubo commented:
@tapasweni-pathak I did "edit" and what I see is only:

![commit](https://cloud.githubusercontent.com/assets/7673323/8638888/0d2e40e2-28c9-11e5-9c0f-b70adef6b11c.png)

no diffs...

@exploreshaifali thank you for your help! I think I understand what the problem is. Next time I will try to branch only from the develop branch.

tapasweni-pathak commented:
@vubo I checked. After this you have to git commit --amend. 

It is not the right use of git rebase, but it used to show the diff. I don't know why it is not showing now. It opens up the commit, but shows the changes files and everything else but not the diffs.

Leave it then. You got the problem, so can you fix it now?

vubo commented:
@tapasweni-pathak if I do git commit --amend, I see:

![image](https://cloud.githubusercontent.com/assets/7673323/8727010/aa8fc7ea-2bdd-11e5-83c8-234680b3ab01.png)

So this PR can be merged, but in the feature I will try to avoid such mistakes.

tapasweni-pathak commented:
No, that's what I said, it is not showing what it used to. So this won't help.

If you got the problem, I was asking you to fix that, the way you are thinking of. 

Okay, I'll merge this, then, np.

vubo commented:
@tapasweni-pathak I do not know how to fix this PR...but I will try to avoid this confused things.

tapasweni-pathak commented:
:+1:
